src/elm/View/Graph.elm

```diff
@@ -208,17 +245,65 @@ allDates dataPoints =
         |> List.sort
 
 
-dateRangeStart : List Data.LineChartDatum -> Float
-dateRangeStart dataPoints =
+dateRangeStart : Data.SectionId -> List Data.LineChartDatum -> Float
+dateRangeStart id dataPoints =
     List.head (allDates dataPoints)
         |> Maybe.withDefault (toFloat 883612800000)
+        |> floorDate id
 
 
-dateRangeEnd : List Data.LineChartDatum -> Float
-dateRangeEnd dataPoints =
+dateRangeEnd : Data.SectionId -> List Data.LineChartDatum -> Float
+dateRangeEnd id dataPoints =
     List.reverse (allDates dataPoints)
         |> List.head
         |> Maybe.withDefault (toFloat 1672531200000)
+        |> ceilingDate id
+
+
+oneYearInMiliseconds : Int
```

Comment:
I think milis needs to be millis everywhere. 2 l's but otherwise this looks great.